tests/src/unit-alt-string.cpp

```diff
@@ -28,7 +28,7 @@ class alt_string
   public:
     using value_type = std::string::value_type;
 
-    static constexpr auto npos = static_cast<std::size_t>(-1);
+    static constexpr auto npos = (std::numeric_limits<std::size_t>::max)();
```

Comment:
I think we can also use the `unknown_size()` function here and on there usages of `std::numeric_limits<std::size_t>::max`

i.e.

`nlohmann::detail::unknown_size()`

Reply:
Yes, but I don't want to rely too much on content from the `detail` namespace as it is allowed to be changed any time.